router/core/router.go

```diff
@@ -1707,6 +1709,15 @@ func WithClientHeader(cfg config.ClientHeader) Option {
 	}
 }
 
+func WithCacheWarmupConfig(cfg *config.CacheWarmupConfiguration) Option {
+	return func(r *Router) {
+		if cfg.Workers <= 0 {
```

Comment:
By design, we never do config validation or default handling in `With` methods. Please do it centrally in `NewRouter`

Reply:
We must validate this here because the central validation would not work as we're unable to use it in testenv.